libgstc/javascript/libgstc.js

```diff
@@ -0,0 +1,598 @@
+/*
+ * GStreamer Daemon - gst-launch on steroids
+ * JavaScript client library abstracting gstd interprocess communication
+ *
+ * Copyright (c) 2015-2020 RidgeRun, LLC (http://www.ridgerun.com)
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/** GstClient - GstdClient Class */
+class GstdClient {
+
+  /**
+   * GstdClient Construtor.
+   *
+   * @param {String} ip.
+   * @param {Number} port.
+   *
+   * @throws {GstcErrorCode}
+  */
+  constructor(ip='http://localhost',port=5000) {
+    this.ip = ip;
+    this.port = port;
+  }
+
+  /**
+   * Send Command.
+   *
+   * @param {String} url.
+   * @param {Array} request.
+   *
+   * @throws {GstdError} Error is triggered when Gstd IPC fails.
+   * @throws {GstcError} Error is triggered when GstClient fails.
+   *
+   * @return {JSON} Response from Gstd.
```

Comment:
Is JSON the appropriate name? I think you return the object and not the JSON.

Reply:
You are right. It is an object